src/IssueViz.Security.UnitTests/Hotspots/HotspotMatcherTests.cs

```diff
@@ -30,6 +36,188 @@ public class HotspotMatcherTests
+            .Should()
+            .BeTrue();
+    }
+    
+    [TestMethod]
+    public void IsMatch_NoHashSameLine_ReturnsTrue()
+    {
+        const string ruleId = "rule1";
+        const string filePath = "A:\\ny\\p\\ath";
+        const string serverPath = "p\\ath";
+        var testSubject = CreateTestSubject();
+
+        testSubject.IsMatch(CreateLocalHotspot(ruleId, "message1", "abchash", filePath, 10),
+                CreateServerHotspot(ruleId, "message2", null, serverPath, 10))
+            .Should()
+            .BeTrue();
+    }
+    
+    [TestMethod]
+    public void IsMatch_SameIssueDifferentHash_ReturnsTrue()
+    {
+        const string ruleId = "rule1";
+        const string filePath = "A:\\ny\\p\\ath";
+        const string serverPath = "p\\ath";
+        var testSubject = CreateTestSubject();
+
+        testSubject.IsMatch(CreateLocalHotspot(ruleId, "message1", "abchash", filePath, 10),
+                CreateServerHotspot(ruleId, "message1", "defhash", serverPath, 10))
+            .Should()
+            .BeTrue();
+    }
+    
+    [TestMethod]
+    public void IsMatch_SameFileAndRuleDifferentIssue_ReturnsFalse()
+    {
+        const string ruleId = "rule1";
+        const string filePath = "A:\\ny\\p\\ath";
+        const string serverPath = "p\\ath";
+        var testSubject = CreateTestSubject();
+
+        testSubject.IsMatch(CreateLocalHotspot(ruleId, "message1", "abchash", filePath, 10),
+                CreateServerHotspot(ruleId, "message2", "defhash", serverPath, 20))
+            .Should()
+            .BeFalse();
+    }
+    
+    [TestMethod]
+    public void IsMatch_SameFileAndRuleNullHashDifferentIssue_ReturnsFalse()
+    {
+        const string ruleId = "rule1";
+        const string filePath = "A:\\ny\\p\\ath";
+        const string serverPath = "p\\ath";
+        var testSubject = CreateTestSubject();
+
+        testSubject.IsMatch(CreateLocalHotspot(ruleId, "message1", "hash", filePath, 10),
+                CreateServerHotspot(ruleId, "message2", null, serverPath, 20))
+            .Should()
+            .BeFalse();
+    }
+    
+    [TestMethod]
+    public void IsMatch_SameHashDifferentRest_ReturnsTrue()
+    {
+        const string ruleId = "rule1";
+        const string filePath = "A:\\ny\\p\\ath";
+        const string serverPath = "p\\ath";
+        const string lineHash = "abchash";
+        var testSubject = CreateTestSubject();
+
+        testSubject.IsMatch(CreateLocalHotspot(ruleId, "message1", lineHash, filePath, 10),
+                CreateServerHotspot(ruleId, "message2", lineHash, serverPath, 20))
+            .Should()
+            .BeTrue();
+    }
+
+    [TestMethod]
+    public void IsMatch_DifferentRuleId_ReturnsFalse()
+    {
+        const string message = "some message";
+        const string filePath = "A:\\ny\\p\\ath";
+        const string serverPath = "p\\ath";
+        const string lineHash = null;
```

Comment:
It would be better to have matching non-null line hash value in this test since a null server line hash is treated differently.